grayscale.py

```diff
@@ -1,764 +1,549 @@
-        self.key = key
-        self.mirrorX = mirrorX
-        self.mirrorY = mirrorY
-
-    @micropython.native
-    def getFrame(self):
-        return self.currentFrame
+from machine import Pin, SPI, idle
+from os import stat
+from utime import sleep_ms, ticks_diff, ticks_ms, ticks_us, sleep_us
+import _thread
+from ssd1306 import SSD1306_SPI
+from thumbyButton import buttonA, buttonB, buttonU, buttonD, buttonL, buttonR
+from thumbySprite import Sprite as _Sprite
 
-    @micropython.native
-    def setFrame(self, frame):
-        if(frame >= 0 and (self.currentFrame is not frame % (self.frameCount))):
-            self.currentFrame = frame % (self.frameCount)
-            offset=self.bitmapByteCount*self.currentFrame
-            if type(self.bitmapSource1)==str:
-                self.file1.seek(offset)
-                self.file1.readinto(self.bitmap1)
-                #f.close()
-            elif type(self.bitmapSource1)==bytearray:
-                self.bitmap1 = memoryview(self.bitmapSource1)[offset:offset+self.bitmapByteCount]
-            if type(self.bitmapSource2)==str:
-                self.file2.seek(offset)
-                self.file2.readinto(self.bitmap2)
-                #f.close()
-            elif type(self.bitmapSource2)==bytearray:
-                self.bitmap2 = memoryview(self.bitmapSource2)[offset:offset+self.bitmapByteCount]
-
-
-# You can't use const() to define class properties, so we'll define them here
-
-# The times below are calculated using phase 1 and phase 2 pre-charge periods of
-# 1 clock.
+# The times below are calculated using phase 1 and phase 2 pre-charge
+# periods of 1 clock.
 # Note that although the SSD1306 datasheet doesn't state it, the 50
-# clocks period per row _is_ a constant (datasheets for similar controllers from
-# the same manufacturer state this).
-# 530kHz is taken to be the highest nominal clock frequency. The calculations
-# shown provide the value in seconds, which can be multiplied by 1e6 to provide
-# a microsecond value.
-_Grayscale_pre_frame_time_us    = const( 785)     # 8 rows: ( 8*(1+1+50)) / 530e3 seconds
-_Grayscale_frame_time_us        = const(4709)     # 48 rows: (49*(1+1+50)) / 530e3 seconds
-
-_Grayscale_ThreadState_Starting   = const(0)
-_Grayscale_ThreadState_Stopped    = const(1)
-_Grayscale_ThreadState_Running    = const(2)
-_Grayscale_ThreadState_Stopping   = const(3)
-
-_Grayscale_StateIndex_State       = const(0)
-_Grayscale_StateIndex_CopyBuffs   = const(1)
-_Grayscale_StateIndex_PendingCmd  = const(2)
-_Grayscale_StateIndex_ContrastChng= const(3)
-
+# clocks period per row _is_ a constant (datasheets for similar
+# controllers from the same manufacturer state this).
+# 530kHz is taken to be the highest nominal clock frequency. The
+# calculations shown provide the value in seconds, which can be
+# multiplied by 1e6 to provide a microsecond value.
+# Both values are in microseconds.
+_PRE_FRAME_TIME = const(785) # 8 rows: (8*(1+1+50)) / 530e3 seconds
```

Comment:
The inconsistent naming styles with the others variables with some capitalised and some other vars not broke me. Sorry. Not cool of me, really.
There is wisdom in this recommendation. I've added _US suffice, but I'm not sure if you prefer _uS? This would be a case where breaking the naming would make total sense. There are too many meanings to US so making it look odd adds clarity! Let me know your preference, and I'l amend however you prefer.

Reply:
Which vars were capitalised? Do you mean BLACK, LIGHTGRAY etc.? Hmm, tbh I think I'd prefer them to have been lowercase, but I know the uppercase for constants thing isn't uncommon (I guess from the use of uppercase in C defines).
Either way in the previous PR I didn't change BLACK, LIGHTGRAY etc. to minimise changes. Maybe it could be argued that those are for external use and so don't need to follow the same naming rules as these timing values?

I've definitely come up against the whole 'rest of the name is uppercase but the suffix looks weird in uppercase' thing before, probably with US/uS. Although I agree _US isn't ideal, I think context should make it clear enough.

Reply:
I actually meant:
```
_Grayscale_frame_time_us
vs
_Grayscale_ThreadState_Starting
```
Which I admit was a subtle difference that I addressed with a sledge hammer.

Happy to change anything back in a moments notice.